container-scan/action.yaml

```diff
@@ -69,14 +59,123 @@ inputs:
+  aws-secret-access-key:
+    required: false
+    description: AWS Secret Access Key
+
+outputs:
+  stdout:
+    description: The complete output from the Wiz CLI scan
+    value: ${{ steps.scan.outputs.stdout }}
+  has_findings:
+    description: Whether the scan found any issues
+    value: ${{ steps.scan.outputs.has_findings }}
+  scan_status:
+    description: Status of the scan (success/failure)
+    value: ${{ steps.scan.outputs.scan_status }}
+  vulnerabilities:
+    description: Summary of vulnerability counts
+    value: ${{ steps.scan.outputs.vulnerabilities }}
+  critical_count:
+    description: Count of critical vulnerabilities
+    value: ${{ steps.scan.outputs.critical_count }}
+  high_count:
+    description: Count of high vulnerabilities
+    value: ${{ steps.scan.outputs.high_count }}
+  medium_count:
+    description: Count of medium vulnerabilities
+    value: ${{ steps.scan.outputs.medium_count }}
+  low_count:
+    description: Count of low vulnerabilities
+    value: ${{ steps.scan.outputs.low_count }}
+  scan_url:
+    description: URL to the scan report
+    value: ${{ steps.scan.outputs.scan_url }}
+  image_id:
+    description: The Docker image ID (SHA256)
+    value: ${{ steps.docker-metadata.outputs.image_id }}
+  image_size:
+    description: The Docker image size in bytes
+    value: ${{ steps.docker-metadata.outputs.image_size }}
+  image_with_tag:
+    description: The full image name with tag
+    value: ${{ steps.docker-build.outputs.image-with-tag }}
 
 runs:
   using: composite
   steps:
     - name: Checkout Repository
-      if: inputs.checkout-repo == 'true'
       uses: actions/checkout@v4
 
+    - name: Download Wiz CLI
+      run: |
+        curl -o wizcli https://downloads.wiz.io/wizcli/latest/wizcli-linux-amd64
+        chmod +x wizcli
+        sudo mv wizcli /usr/local/bin/
+      shell: bash
+
+    - name: Authenticate with Wiz
+      run: wizcli auth --id "$WIZ_CLIENT_ID" --secret "$WIZ_CLIENT_SECRET"
+      env:
+        WIZ_CLIENT_ID: ${{ inputs.wiz-client-id }}
+        WIZ_CLIENT_SECRET: ${{ inputs.wiz-client-secret }}
+      shell: bash
+
+    - name: Setup Tools
+      uses: open-turo/action-setup-tools@v2
+
+    - name: Run Container Setup Commands
+      if: inputs.container-setup-commands != ''
+      shell: bash
+      run: |
+        echo "Running container setup commands after checkout"
+        ${{ inputs.container-setup-commands }}
+      env:
+        # AWS Credentials
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-secret-access-key }}
+        # Artifactory Credentials
+        ARTIFACTORY_USERNAME: ${{ inputs.artifactory-username }}
+        ARTIFACTORY_AUTH_TOKEN: ${{ inputs.artifactory-auth-token }}
+        # Other common secrets
+        MAX_MIND_LICENSE_KEY: ${{ inputs.MAX_MIND_LICENSE_KEY }}
+        DOCKER_USERNAME: ${{ inputs.dockerhub-user }}
+        DOCKER_PASSWORD: ${{ inputs.dockerhub-password }}
```

Comment:
To not couple things here to all these tools, would it make more sense to have a generic input `container-setup-env` where users can specify the things they need

Reply:
Like `input.secrets` and `input.build-args` below